test/services/nexboard/MockServer.js

```diff
@@ -0,0 +1,123 @@
+	app.use(bodyParser.urlencoded({ extended: true })); // support encoded bodies
+	app.url = url;
+
+	const uris = {
+		postProject: `/${uri}projects`,
+		getProject: `/${uri}projects/:id`,
+		findProject: `/${uri}projects`,
+		postBoards: `/${uri}boards`,
+		getProjectBoards: `/${uri}projects/:projectId/boards`,
+		getBoard: `/${uri}boards/:id`,
+	};
+
+	app.get('/ping', (req, res) => {
+		res.json({
+			pong: 'true',
+		});
+	});
+
+	app.post(uris.postProject, (req, res) => {
+		res.json({
+			id: '3414',
+			title: req.body.title || 'Neues Nexboard Projekt',
+			description: req.body.description || 'Hier werden alle Nexboards für diese Lerneinheit gesammelt',
+		});
+	});
+
+	app.get(uris.getProject, (req, res) => {
+		res.json({
+			id: req.params.id,
+			title: 'Neues Nexboard Projekt',
+			description: 'Hier werden alle Nexboards für diese Lerneinheit gesammelt',
+			ownerId: '123',
+			boardIds: [],
+		});
+	});
+
+	app.get(uris.findProject, (req, res) => {
+		res.json([3414]);
+	});
+
+	app.post(uris.postBoards, (req, res) => {
+		res.json({
+			creationDate: 1555423970889,
+			lastModified: 1555423970889,
+			vcSessionDate: null,
+			id: '14212',
+			title: req.body.title || 'Neues Nexboard Board',
+			description: req.body.description || 'Ein digitales Whiteboard',
+			projectId: req.body.projectId || null,
+			isTemplate: false,
+			publicLink: `${nexUrl}/app/client/pub/14212/766u0758-n138-0qx1-m1z3-0711ru670706`,
+			preview: `${nexUrl}/screenshots/14212?date=1555423972&width=300&height=210`,
+			image: `${nexUrl}/screenshots/14212?date=1555423972&width=1900&height=1200`,
+		});
+	});
+
+	app.get(uris.getProjectBoards, (req, res) => {
+		if (!req.params.projectId) return res.status(400).send('Could not retrieve projects');
+
+		return res.json([{
+			id: '14212',
+			key: 'S56abfefa',
+			title: 'Neues Nexboard Board',
+			projectId: req.params.projectId || 3414,
+			description: 'Ein digitales Whiteboard',
+			creationDate: 1555423970889,
+			lastModified: 1555423970889,
+			isTemplate: false,
+			meta: null,
+			isPublic: false,
+			publicLink: `${nexUrl}/app/client/pub/14212/766u0758-n138-0qx1-m1z3-0711ru670706`,
+			preview: `${nexUrl}/screenshots/14212?date=1555424066&width=300&height=210`,
+			image: `${nexUrl}/screenshots/14212?date=1555424066&width=1900&height=1200`,
+		}]);
+	});
+
+	app.get(uris.getBoard, (req, res) => {
+		res.json({
+			id: req.params.id,
+			key: 'S56abfefa',
+			title: 'Neues Nexboard Board',
+			projectId: req.query.projectId || 3414,
+			description: 'Ein digitales Whiteboard',
+			creationDate: 1555423970889,
+			lastModified: 1555423970889,
+			isTemplate: false,
+			meta: null,
+			isPublic: false,
+			publicLink: `${nexUrl}/app/client/pub/14212/766u0758-n138-0qx1-m1z3-0711ru670706`,
+			preview: `${nexUrl}/screenshots/14212?date=1555424066&width=300&height=210`,
+			image: `${nexUrl}/screenshots/14212?date=1555424066&width=1900&height=1200`,
+		});
+	});
+
+	const server = app.listen(port);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['port' is not defined.](https://app.codacy.com/manual/schul-cloud/schulcloud-server/pullRequest?prid=5110468)